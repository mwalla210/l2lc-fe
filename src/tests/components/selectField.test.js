import React from 'react'
import renderer from 'react-test-renderer'
import SelectField from '../../components/selectField'
import {observable} from 'mobx'

const defaultOptions = {
  valid: true,
  disabled: false,
  id: 'id',
  value: 'value',
  onChange: jest.fn(),
  onBlur: jest.fn(),
  focus: true,
  options:observable([
    {
      disabled: true,
      title: 'title'
    },
    {
      title: 'title'
    }
  ])
}

const alternateOptions = {
  valid: false,
  disabled: false,
  id: 'id',
  value: 'value',
  onChange: jest.fn(),
  onBlur: jest.fn(),
  focus: true,
  options:observable([
    {
      disabled: true,
      title: 'title'
    },
    {
      title: 'title'
    }
  ])
}

describe('SelectField', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <SelectField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguements', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <SelectField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
