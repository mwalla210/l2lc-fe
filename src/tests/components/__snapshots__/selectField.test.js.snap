// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Renders with snapshot 1`] = `
<select
  autoFocus={true}
  className="form-control"
  disabled={false}
  id="id"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  style={
    Object {
      "borderColor": "",
    }
  }
  value="value"
>
  <option
    disabled={true}
  >
    title
  </option>
  <option>
    title
  </option>
</select>
`;

exports[`SelectField Renders with snapshot, alternate arguements 1`] = `
<select
  autoFocus={true}
  className="form-control"
  disabled={false}
  id="id"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  style={
    Object {
      "borderColor": "orange",
    }
  }
  value="value"
>
  <option
    disabled={true}
  >
    title
  </option>
  <option>
    title
  </option>
</select>
`;
